File: 0056-merge-intervals/0056-merge-intervals.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """
        intervals.sort(key=lambda x: x[0]) 
        start = intervals[0][0]
        end = intervals[0][1]

        answer=[]

        for i in range(1, len(intervals)):
           
            s = intervals[i][0]
            e = intervals[i][1]

            if end >= s:
                end = max(end, e)

            else:
                answer.append([start, end])
                start = s
                end = e

            
        answer.append([start, end])
        return answer

File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> a[0] - b[0]);

        int start = intervals[0][0];
        int end = intervals[0][1];
        ArrayList<int []> answer = new ArrayList<>();
        for(int i=1; i<intervals.length; i++){
            int s = intervals[i][0];
            int e = intervals[i][1];
            if(end>=s){
                end = Math.max(end, e);
                
            }
            else{
                answer.add(new int[]{start,end});
                start = s;
                end = e;
            }
        }
        answer.add(new int[]{start,end});
        int ans[][] = answer.toArray(new int[0][]);
        return ans;
        
    }
}